### DISCUSSION TITLE
graphql/envelop/plugins/use-validation-cache

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/envelop#1747] [Docs Discussions]
# graphql/envelop/plugins/use-validation-cache

GraphQL Envelop Documentation

https://the-guild.dev/graphql/envelop/plugins/use-validation-cache

**Comment by SiNONiMiTY** [comment]
This plugin returns an error

```
SHA1.hash is not a function
    err: {
      "type": "TypeError",
      "message": "SHA1.hash is not a function",
      "stack":
          TypeError: SHA1.hash is not a function
              at getSchemaHash
```

Seems that the hashing function library used is not working properly. Why not use the node:crypto module for SHA1 hashing...

**Reply by ardatan** [reply]
We don't use Node crypto module because this library is not used only on Node. Could you create a reproduction on CodeSandbox or StackBlitz?

**Reply by SiNONiMiTY** [reply]
I tried creating a reproduction but cannot replicate the SHA1.hash is not a function error. Will investigate further and report the results here.

**Reply by SiNONiMiTY** [reply]
@ardatan was able to replicate the issue by imitating my TS project settings and manually adding sha1-es as a dependency.

Please see the replication @ https://stackblitz.com/edit/node-dap7mf?file=src/index.mts

Maybe an issue with the sha1 library not being compatible with modules? I am not sure.
